RecordBatch panics if creates measurement before InstallNewPipeline

The code below will panic with `interface conversion: metric.SyncImpl is *internal.syncImpl, not *metric.syncInstrument`.

```go
package main

import (
	"context"
	"log"

	"go.opentelemetry.io/otel/api/global"
	"go.opentelemetry.io/otel/api/metric"
	"go.opentelemetry.io/otel/exporters/metric/stdout"
)

func main() {
	log.SetFlags(0)
	meter := global.MeterProvider().Meter("example/ping")
	counter := metric.Must(meter).NewInt64Counter("example.counter")

	pusher, err := stdout.InstallNewPipeline(stdout.Config{})
	if err != nil {
		log.Fatal(err)
	}
	defer pusher.Stop()

	n := counter.Measurement(1)
	meter.RecordBatch(context.Background(), nil, n)
}
```

Should I keep to install the exporter before creating any instruments?

### environment

- Go: 1.14
- GOOS: darwin
- GOARCH: amd64
- otel: v0.4.3

Full output reproducing the above error with the same sample program:

```
go run test.go
{"time":"2020-04-29T13:05:12.257716346-07:00","updates":null}
panic: interface conversion: metric.SyncImpl is *internal.syncImpl, not *metric.syncInstrument

goroutine 1 [running]:
go.opentelemetry.io/otel/sdk/metric.(*SDK).RecordBatch(0xc0000b8000, 0x574fc0, 0xc000014110, 0x0, 0x0, 0x0, 0xc00000c260, 0x1, 0x1)
	/home/tyler/go/pkg/mod/go.opentelemetry.io/otel@v0.4.3/sdk/metric/sdk.go:473 +0x192
go.opentelemetry.io/otel/api/metric/registry.(*uniqueInstrumentMeterImpl).RecordBatch(0xc00000c200, 0x574fc0, 0xc000014110, 0x0, 0x0, 0x0, 0xc00000c260, 0x1, 0x1)
	/home/tyler/go/pkg/mod/go.opentelemetry.io/otel@v0.4.3/api/metric/registry/registry.go:58 +0x94
go.opentelemetry.io/otel/api/metric.(*wrappedMeterImpl).RecordBatch(0xc00000c240, 0x574fc0, 0xc000014110, 0x0, 0x0, 0x0, 0xc00000c260, 0x1, 0x1)
	/home/tyler/go/pkg/mod/go.opentelemetry.io/otel@v0.4.3/api/metric/sdkhelpers.go:134 +0x93
go.opentelemetry.io/otel/api/global/internal.(*meter).RecordBatch(0xc0000621e0, 0x574fc0, 0xc000014110, 0x0, 0x0, 0x0, 0xc00000c260, 0x1, 0x1)
	/home/tyler/go/pkg/mod/go.opentelemetry.io/otel@v0.4.3/api/global/internal/meter.go:331 +0xa5
main.main()
	/home/tyler/go/src/test/test.go:24 +0x2ba
exit status 2
```

--- a/api/global/internal/meter_test.go
+++ b/api/global/internal/meter_test.go
@@ -20,7 +20,6 @@ import (
 	"errors"
 	"io"
 	"io/ioutil"
-	"log"
 	"testing"
 
 	"github.com/stretchr/testify/require"
@@ -405,7 +404,7 @@ func TestRecordBatchRealSDK(t *testing.T) {
 		DoNotPrintTime: true,
 	})
 	if err != nil {
-		log.Fatal(err)
+		t.Fatal(err)
 	}
 	global.SetMeterProvider(pusher)
 

--- a/api/global/internal/meter_test.go
+++ b/api/global/internal/meter_test.go
@@ -15,10 +15,12 @@
 package internal_test
 
 import (
+	"bytes"
 	"context"
 	"errors"
 	"io"
 	"io/ioutil"
+	"log"
 	"testing"
 
 	"github.com/stretchr/testify/require"
@@ -359,4 +361,57 @@ func TestImplementationIndirection(t *testing.T) {
 
 	_, ok = ival.(*metrictest.Async)
 	require.True(t, ok)
+}
+
+func TestRecordBatchMock(t *testing.T) {
+	internal.ResetForTest()
+
+	meter := global.MeterProvider().Meter("builtin")
+
+	counter := Must(meter).NewInt64Counter("test.counter")
+
+	meter.RecordBatch(context.Background(), nil, counter.Measurement(1))
+
+	mock, provider := metrictest.NewProvider()
+	global.SetMeterProvider(provider)
+
+	meter.RecordBatch(context.Background(), nil, counter.Measurement(1))
+
+	require.EqualValues(t,
+		[]measured{
+			{
+				Name:        "test.counter",
+				LibraryName: "builtin",
+				Labels:      asMap(),
+				Number:      asInt(1),
+			},
+		},
+		asStructs(mock.MeasurementBatches))
+}
+
+func TestRecordBatchRealSDK(t *testing.T) {
+	internal.ResetForTest()
+
+	meter := global.MeterProvider().Meter("builtin")
+
+	counter := Must(meter).NewInt64Counter("test.counter")
+
+	meter.RecordBatch(context.Background(), nil, counter.Measurement(1))
+
+	var buf bytes.Buffer
+
+	pusher, err := stdout.InstallNewPipeline(stdout.Config{
+		Writer:         &buf,
+		DoNotPrintTime: true,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	global.SetMeterProvider(pusher)
+
+	meter.RecordBatch(context.Background(), nil, counter.Measurement(1))
+	pusher.Stop()
+
+	require.Equal(t, `{"updates":[{"name":"test.counter","sum":1}]}
+`, buf.String())
 }

--- a/internal/metric/mock.go
+++ b/internal/metric/mock.go
@@ -169,7 +169,7 @@ func (m *MeterImpl) RecordBatch(ctx context.Context, labels []core.KeyValue, mea
 	for i := 0; i < len(measurements); i++ {
 		m := measurements[i]
 		mm[i] = Measurement{
-			Instrument: m.SyncImpl().(*Sync),
+			Instrument: m.SyncImpl().Implementation().(*Sync),
 			Number:     m.Number(),
 		}
 	}

--- a/sdk/metric/sdk.go
+++ b/sdk/metric/sdk.go
@@ -470,7 +470,7 @@ func (m *SDK) RecordBatch(ctx context.Context, kvs []core.KeyValue, measurements
 	// ordered labels.
 	var labelsPtr *label.Set
 	for i, meas := range measurements {
-		s := meas.SyncImpl().(*syncInstrument)
+		s := meas.SyncImpl().Implementation().(*syncInstrument)
 
 		h := s.acquireHandle(kvs, labelsPtr)
 

